app/src/main/java/com/github/sdp/mediato/MainActivity.java

```diff
@@ -29,24 +30,30 @@ public class MainActivity extends AppCompatActivity implements FragmentSwitcher
 
   ActivityMainBinding binding;
   MyProfileFragment myProfileFragment;
+  ReadOnlyProfileFragment readOnlyProfileFragment;
   SearchFragment searchFragment;
   ExploreFragment exploreFragment;
 
-  private ProfileViewModel currentUserViewModel;
-  private ProfileViewModel otherUsersViewModel;
+  private MyProfileViewModel myProfileViewModel;
+  private ReadOnlyProfileViewModel readOnlyProfileViewModel;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     binding = ActivityMainBinding.inflate(getLayoutInflater());
     setContentView(binding.getRoot());
 
-    // Choose the default fragment that opens on creation of the MainActivity
-    setDefaultFragment();
-
     // Initialize the ViewModels
-    currentUserViewModel = new ViewModelProvider(this).get(ProfileViewModel.class);
-    otherUsersViewModel = new ViewModelProvider(this).get(ProfileViewModel.class);
+    myProfileViewModel = new ViewModelProvider(this).get(MyProfileViewModel.class);
+    readOnlyProfileViewModel = new ViewModelProvider(this).get(ReadOnlyProfileViewModel.class);
```

Comment:
Is there a reason why the ViewModel is created in this Activity and not in their fragment ? (I know that you did not introduce this change yourself, but it might be a good idea to clean it.)